framework/include/base/MeshGeneratorSystem.h

```diff
@@ -0,0 +1,199 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "InputParameters.h"
+#include "PerfGraphInterface.h"
+
+#include "libmesh/parallel_object.h"
+#include "libmesh/mesh_base.h"
+
+class MooseApp;
+class MeshGeneratorMesh;
+class MeshGenerator;
+
+/**
+ * System that manages MeshGenerators.
+ *
+ * To be owned by the MooseApp.
+ */
+class MeshGeneratorSystem : public PerfGraphInterface, public libMesh::ParallelObject
+{
+public:
+  MeshGeneratorSystem(MooseApp & app);
+
+  /**
+   * Execute and clear the Mesh Generators data structure
```

Comment:
add a short note that they are previously ordered using dependency resolution

Reply:
The final ordering is done within `executeMeshGenerators()`, there is no previous execution ordering.

There's a separate ordering during _construction_, but that's a different method which has its own detailed explanation

Reply:
hence why we need a note about ordering here too

Reply:
Functions should be written so that they're modular and independent. If you want to suggest something I'll add it but I'm outta juice

Reply:
I ll be human and take that into account ^^